pkg/model/devrc.go

```diff
@@ -0,0 +1,187 @@
+package model
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/okteto/okteto/pkg/log"
+	yaml "gopkg.in/yaml.v2"
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// Dev represents a development container
+type DevRC struct {
+	Labels               Labels                `json:"labels,omitempty" yaml:"labels,omitempty"`
+	Annotations          Annotations           `json:"annotations,omitempty" yaml:"annotations,omitempty"`
+	Context              string                `json:"context,omitempty" yaml:"context,omitempty"`
+	Namespace            string                `json:"namespace,omitempty" yaml:"namespace,omitempty"`
+	Environment          Environment           `json:"environment,omitempty" yaml:"environment,omitempty"`
+	Secrets              []Secret              `json:"secrets,omitempty" yaml:"secrets,omitempty"`
+	Sync                 Sync                  `json:"sync,omitempty" yaml:"sync,omitempty"`
+	Resources            ResourceRequirements  `json:"resources,omitempty" yaml:"resources,omitempty"`
+	PersistentVolumeInfo *PersistentVolumeInfo `json:"persistentVolume,omitempty" yaml:"persistentVolume,omitempty"`
+	Timeout              time.Duration         `json:"timeout,omitempty" yaml:"timeout,omitempty"`
+}
+
+// Get returns a Dev object from a given file
+func GetRc(devPath string) (*DevRC, error) {
+	b, err := ioutil.ReadFile(devPath)
+	if err != nil {
+		return nil, err
+	}
+
+	dev, err := ReadRC(b)
+	if err != nil {
+		log.Warning("ignoring developer overwrites defined in %s: %s", devPath, err.Error())
+		return nil, err
+	}
+
+	return dev, nil
+}
+
+// Read reads an okteto manifests
+func ReadRC(bytes []byte) (*DevRC, error) {
+	dev := &DevRC{}
+
+	if bytes != nil {
+		if err := yaml.UnmarshalStrict(bytes, dev); err != nil {
+			if strings.HasPrefix(err.Error(), "yaml: unmarshal errors:") {
+				var sb strings.Builder
+				_, _ = sb.WriteString("Invalid manifest:\n")
+				l := strings.Split(err.Error(), "\n")
+				for i := 1; i < len(l); i++ {
+					e := strings.TrimSuffix(l[i], "in type model.Dev")
```

Comment:
how does this look like? I don't think we should include `model.Dev` in the error, since the typical user doesn't understand the internals of the CLI. I think it's better to talk about the file, which line is wrong, and point them to the docs.

Reply:
We have similar code for model.dev, could we reuse it, since this is a subset?